src/parsing.jl

```diff
@@ -119,7 +121,11 @@ By default, this uses any existing `parse` method for parsing a value of type `T
 You can implement default PostgreSQL-specific parsing for a given type by overriding
 `pqparse`.
 """
-Base.parse(::Type{T}, pqv::PQValue) where {T} = pqparse(T, string_view(pqv))
+Base.parse(::Type{T}, pqv::PQValue) where {T, R} = pqparse(T, string_view(pqv))
+# Base.parse(::Type{T}, pqv::PQValue{R, false}) where {T, R} = pqparse(T, string_view(pqv))
+
+# Base.parse(::Type{T}, pqv::PQValue{R, true}) where {T, R} = bswap(unsafe_load(Ptr{T}(data_pointer(pqv))))
+Base.parse(::Type{T}, pqv::PQValue{R, true}) where {T <: Integer, R} = bswap(unsafe_load(Ptr{T}(data_pointer(pqv))))
```

Comment:
The size of the pointer should be determined by `R`, not `T`. 

We should also make this `pgparse`, and have a fallback `Base.parse` that parses the binary data into the default type, then converts to the desired type (this would let us continue to ask for, e.g., 64-bit ints when the data is 32-bit ints).

Reply:
Do we definitely want `bswap` here rather than `ntoh`?

Reply:
> Do we definitely want bswap here rather than ntoh? (I think this currently assumes that the host system is little endian)

It probably should be `ntoh`

If i have the right parts of the docs: 
https://www.postgresql.org/docs/9.3/protocol-overview.html#PROTOCOL-FORMAT-CODES

> Binary representations for integers use network byte order (most significant byte first).


Julia doesn't actually run on anything big endian, several places in base use `bswap` and `ntoh` incorrectly.

Reply:
> The size of the pointer should be determined by R, not T.

I'm not sure how exactly to do this. R is just a number that needs to be converted to a type. And the type conversion has already been thrown out. 

Also, isn't T the proper type that was determined by R?

Reply:
> Base.parse(::Type{T}, pqv::PQValue{R, true}) where {T <: Integer, R} = bswap(unsafe_load(Ptr{T}(data_pointer(pqv))))

With the `Ptr{T}` conversion you're asserting that the pointer points to data of `sizeof(T)` bytes, then `unsafe_load` will grab that many bytes and interpret them as `T`. But `T` is just what the user is asking for, it does not need to correspond to the type of the data returned by PostgreSQL. `R` is what determines the type of the data pointed to by `data_pointer(pqv)` and, in the case of fixed-length types like integers, determines the number of bytes that should be read. If you query for a PostgreSQL `int2` and try to parse an `Int64` with the above method, you'll either get incorrect data or a segfault. You _need_ to parse an `Int16` first, and then convert to an `Int64`.

Reply:
Alright, I updated it to use `R`. Though I think the method looks messy. Am I doing something wrong or is this good?

Reply:
I've written what I think should happen at https://github.com/invenia/LibPQ.jl/pull/223/files#r667189710